### PR TITLE
[solarforecast] Add missing @ActionOutput annotation

### PR DESCRIPTION
**Pull request by lolodomo**
Related to #17504

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

File: bundles/org.openhab.binding.solarforecast/src/main/java/org/openhab/binding/solarforecast/internal/actions/SolarForecastActions.java [hunk 1]
```diff
@@ -172,15 +172,15 @@ public void triggerUpdate() {
         }
     }
 
-    public static State getDay(ThingActions actions, LocalDate ld, String... args) {
+    public static QuantityType<Energy> getDay(ThingActions actions, LocalDate ld, String... args) {
```
Comment (lsiepel) [inline bundles/org.openhab.binding.solarforecast/src/main/java/org/openhab/binding/solarforecast/internal/actions/SolarForecastActions.java:175]: Not related to this PR, but the method name is somewhat confusing. its getting the energy state, not the day/datetime.

File: bundles/org.openhab.binding.solarforecast/src/main/java/org/openhab/binding/solarforecast/internal/actions/SolarForecastActions.java [hunk 2]
```diff
@@ -108,10 +108,10 @@ public QuantityType<Power> getPower(
     }
 
     @RuleAction(label = "@text/actionEnergyLabel", description = "@text/actionEnergyDesc")
-    public QuantityType<Energy> getEnergy(
+    public @ActionOutput(name = "getEnergy", type = "QuantityType<Energy>") QuantityType<Energy> getEnergy(
             @ActionInput(name = "start", label = "@text/actionInputDateTimeBeginLabel", description = "@text/actionInputDateTimeBeginDesc") Instant start,
             @ActionInput(name = "end", label = "@text/actionInputDateTimeEndLabel", description = "@text/actionInputDateTimeEndDesc") Instant end,
-            String... args) {
+            @ActionInput(name = "arguments") String... args) {
```
Comment (lsiepel) [inline bundles/org.openhab.binding.solarforecast/src/main/java/org/openhab/binding/solarforecast/internal/actions/SolarForecastActions.java:114]: Shouldn't this be:
```suggestion
            @ActionInput(name = "args") String... args) {
```

Comment (lolodomo) [reply bundles/org.openhab.binding.solarforecast/src/main/java/org/openhab/binding/solarforecast/internal/actions/SolarForecastActions.java:114]: Yes, that would be better.

File: bundles/org.openhab.binding.solarforecast/src/main/java/org/openhab/binding/solarforecast/internal/actions/SolarForecastActions.java (hunk 1 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.solarforecast/src/main/java/org/openhab/binding/solarforecast/internal/actions/SolarForecastActions.java]: Ok, method renamed into `getEnergyOfDay`

Comment (lsiepel) [reply bundles/org.openhab.binding.solarforecast/src/main/java/org/openhab/binding/solarforecast/internal/actions/SolarForecastActions.java]: I think this is much better, but be aware this is a breaking change (users have to adapt their rules). And the readme needs to be adapted too, but that is no real problem.

Comment (lolodomo) [reply bundles/org.openhab.binding.solarforecast/src/main/java/org/openhab/binding/solarforecast/internal/actions/SolarForecastActions.java]: README is now updated.

For the breaking change, a PR in distro is required.

**Pull request merged**